### ISSUE TITLE
Type of product from dictionary is not been taken over in product tab

### ISSUE THREAD

**Issue by joerglott** [Dolibarr/dolibarr#18477]
# Instructions
*This is a template to help you report good issues. You may use [Github Markdown](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/) syntax to format your issue report.*
*Please:*
- *replace the bracket enclosed texts with meaningful information*
- *remove any unused sub-section*

# Bug
Type of product from dictionary is not been taken over in product tab

## Environment
- **Version**: Dolibarr 14.0.0 - Dolibarr 14.0.1 (maybe also in further versions)
- **OS**: Linux Ubuntu 20.04 LTS
- **Web server**: Apache 2.4.41
- **PHP**: PHP 7.4.22
- **Database**: mysql Server 8.0.26
- **URL(s)**: several Instances

## Expected and actual behavior
In the product tab there is existing the field "Type of product" - the informations seems to be from the dictionary "Type of products" and there are the same informations in the base installation. Adding or changing these informations in the dictionary do not affect the list in the product type.

## Steps to reproduce the behavior
1. Go to the dictionary and add an type of product
2. Got to the product page and open the drop down list of the type of product
3. there are only listed the old values and not the added or changed values

## [Attached files](https://help.github.com/articles/issue-attachments) (Screenshots, screencasts, dolibarr.log, debugging informations…)
None

**Comment by cfoellmann** [comment]
I started to look into this.
I am running the current develop branch (15.x-alpha):
It seems the `function getLibFinished()` that returns all set "natures" for products was changed in Feb 2021 (13.x) so it should work.

- I added a new nature to the dictionary. OK

- select it on product/card.php when editing an existing product. OK
- Create a new product and select the new nature. NOT OK -- the select field is not filled by the proper function
- list product/list.php show column nature. OK -- filter select shows all natures and filter can be applied



For me it is possible to add a new nature and then select it on product/card.php when editing a product.
